### PR TITLE
Fix Banner check on filesystem containing special chars

### PR DESCRIPTION
**Pull request by andreaTP**
This is a fix for the issue discovered in Keycloak here: https://github.com/keycloak/keycloak/issues/10954

I'm completely open for guidance in changing the approach for the test.

cc. @pedroigor @DGuhr

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/steps/BannerProcessor.java [hunk 1]
```diff
@@ -114,9 +113,10 @@ private boolean isQuarkusCoreBanner(URL url) throws IOException {
         // We determine whether the banner is the default by checking to see if the jar that contains it also
         // contains this class. This way although somewhat complicated guarantees that any rename of artifacts
         // won't affect the check
-        try (JarFile jarFile = new JarFile(Paths.get(new URI(jarPath)).toFile())) {
+        try {
+            JarFile jarFile = new JarFile(new File(jarPath));
```
Comment (aloubyansky) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/BannerProcessor.java:117]: The `jarFile` needs to be closed. Was there a reason to move it inside the `try`?

Comment (andreaTP) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/BannerProcessor.java]: no, you are right @aloubyansky reverted 👍  good catch!

**Review by aloubyansky — APPROVED** [review]
Approved. Thanks @andreaTP
I am not a fan of committing JARs into the repo but perhaps it's just me. An alternative could be to generate them preparing for the test run. @gsmet do you have any opinion on that?

**Comment by andreaTP** [comment]
@aloubyansky thanks! I'm not a fan of committing jar files either, in this case we are actually committing 2 plain zip files (renamed to `.jar`) and they both only contain one single (different) empty file each.

**Comment by gsmet** [comment]
I think it's OK in theory but my guess is that Windows might not like it. We will see what CI says.
If you can reproduce your problem with `é` or `à`, I would recommend to use these characters to be a bit more portable.

**Comment by andreaTP** [comment]
@gsmet you are right, the CI is failing on Windows 🙁 
Unfortunately the `é` and `à` characters are not triggering the issue.

Can we skip this test on Windows? Any alternative approach?

**Comment by gsmet** [comment]
Let's wait for the full CI report and we will decide from there.

**Comment by gsmet** [comment]
Thanks!

**Comment by Postremus** [comment]
FTR. This introduces a regression, see #24686.

**Comment by gsmet** [comment]
I had to revert this fix and the jar files added in this PR also breaks the Jakarta transformation:
```
jbang alias add --name transform org.eclipse.transformer:org.eclipse.transformer.cli:0.2.0
jbang transform -o core TEMP
```
leads to:
```
[main] ERROR Transformer - Unexpected failure:
java.lang.NullPointerException: name
	at java.base/java.util.Objects.requireNonNull(Objects.java:233)
	at java.base/java.util.zip.ZipEntry.<init>(ZipEntry.java:106)
	at org.eclipse.transformer.action.impl.ContainerActionImpl.apply(ContainerActionImpl.java:281)
	at org.eclipse.transformer.action.impl.ContainerActionImpl.apply(ContainerActionImpl.java:159)
	at org.eclipse.transformer.action.impl.ActionImpl.apply(ActionImpl.java:622)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:103)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.transform(DirectoryActionImpl.java:93)
	at org.eclipse.transformer.action.impl.DirectoryActionImpl.apply(DirectoryActionImpl.java:63)
	at org.eclipse.transformer.Transformer$TransformOptions.transform(Transformer.java:1781)
	at org.eclipse.transformer.Transformer.run(Transformer.java:1874)
	at org.eclipse.transformer.jakarta.JakartaTransformer.main(JakartaTransformer.java:30)
```

**Comment by andreaTP** [comment]
sorry for the trouble!

@gsmet would you like to provide guidance on how to test this?

**Comment by gsmet** [comment]
See the `jbang` lines above. If you just run them at the root of the Quarkus project, you will have the error.

I tried building a proper jar with the `jar` command but it didn't fix the issue and then I gave up :).

**Comment by andreaTP** [comment]
Thanks, I'll take a look next week probably.

**Comment by Postremus** [comment]
Hi @andreaTP,

if you want, I can take this effort over.
I already dug a bit into the banner code in #24690. Solving the jakarta issues shouln't be that big of an additional burden (famous last words..).

Alternativly, if you wnat to continue working on this, I could also just verify that everything works on windows?
Up to you how we proceed with these changes, whatever fits better with your shedule.

**Comment by andreaTP** [comment]
@Postremus I will not look into this until next week, if you want to take over feel free to go ahead. I just needed to fix the special characters thing 🙂 
Ping me if you open a PR otherwise I'll send a message here when I start to work on this 👍

**Pull request merged**